server/test/medium/specs/services/workflow.service.spec.ts

```diff
@@ -0,0 +1,640 @@
+import { Kysely } from 'kysely';
+import { AccessRepository } from 'src/repositories/access.repository';
+import { LoggingRepository } from 'src/repositories/logging.repository';
+import { PluginRepository } from 'src/repositories/plugin.repository';
+import { WorkflowRepository } from 'src/repositories/workflow.repository';
+import { DB } from 'src/schema';
+import { PluginContext, PluginTriggerType } from 'src/schema/tables/plugin.table';
+import { WorkflowService } from 'src/services/workflow.service';
+import { newMediumService } from 'test/medium.factory';
+import { getKyselyDB } from 'test/utils';
+
+let defaultDatabase: Kysely<DB>;
+
+const setup = (db?: Kysely<DB>) => {
+  return newMediumService(WorkflowService, {
+    database: db || defaultDatabase,
+    real: [WorkflowRepository, PluginRepository, AccessRepository],
+    mock: [LoggingRepository],
+  });
+};
+
+beforeAll(async () => {
+  defaultDatabase = await getKyselyDB();
+});
+
+describe(WorkflowService.name, () => {
+  let testPluginId: string;
+  let testFilterId: string;
+  let testActionId: string;
+
+  beforeEach(async () => {
+    // Create a test plugin with filters and actions before each test
+    const pluginRepo = new PluginRepository(defaultDatabase);
+    const result = await pluginRepo.loadPlugin({
+      name: 'test-core-plugin',
+      displayName: 'Test Core Plugin',
+      description: 'A test core plugin for workflow tests',
+      author: 'Test Author',
+      version: '1.0.0',
+      wasm: {
+        path: '/test/path.wasm',
+      },
+      filters: [
+        {
+          name: 'test-filter',
+          displayName: 'Test Filter',
+          description: 'A test filter',
+          supportedContexts: [PluginContext.Asset],
+          schema: undefined,
+        },
+      ],
+      actions: [
+        {
+          name: 'test-action',
+          displayName: 'Test Action',
+          description: 'A test action',
+          supportedContexts: [PluginContext.Asset],
+          schema: undefined,
+        },
+      ],
+    });
+
+    testPluginId = result.plugin.id;
+    testFilterId = result.filters[0].id;
+    testActionId = result.actions[0].id;
+  });
+
+  afterEach(async () => {
+    await defaultDatabase.deleteFrom('plugin').where('id', '=', testPluginId).execute();
```

Comment:
why do you need to create and then delete the same plugin in every test? just create it on ce and leave it there the whole time